src/main/java/eggventory/parsers/ParseAdd.java

```diff
@@ -1,7 +1,93 @@
 package eggventory.parsers;
 
+import eggventory.commands.Command;
+import eggventory.commands.AddCommand;
+import eggventory.enums.CommandType;
+import eggventory.exceptions.BadInputException;
+import eggventory.exceptions.InsufficientInfoException;
+
+
 public class ParseAdd {
-    public ParseAdd(String input) {
 
+    /**
+     * Processes the contents of an add stock command (everything after the words "add" and "stock").
+     * Splits up the input string into an array containing the various attributes of the stock being added.
+     * Ignores leading/trailing whitespace between the first word and subsequent string,
+     * and between all commands' arguments.
+     *
+     * @param input String containing the attributes of the stock.
+     * @return an array consisting of StockType, StockCode, Quantity and Description.
+     * @throws InsufficientInfoException If any of the required attributes is missed out.
+     */
+    private Command processAddStock(String input) throws InsufficientInfoException {
+
+        String[] addInput = input.split(" +", 4); //There are 4 attributes for now.
```

Comment:
I didnt understand this part from earlier code as well. but why are we splitting based on "+" ?